samza-core/src/main/java/org/apache/samza/operators/impl/ControlMessageSender.java

```diff
@@ -29,12 +29,16 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
 
 /**
  * This is a helper class to broadcast control messages to each partition of an intermediate stream
  */
 class ControlMessageSender {
   private static final Logger LOG = LoggerFactory.getLogger(ControlMessageSender.class);
+  private static final Map<SystemStream, Integer> partitions = new ConcurrentHashMap<>();
```

Comment:
s/partitions/partitionCounts maybe?

Reply:
or even partitionCountCache

Reply:
+1

Reply:
Fixed. Changed it to PARTITION_COUNT_CACHE.